Gordon360/Services/ActivityService.cs

```diff
@@ -364,4 +371,19 @@ public void TogglePrivacy(string activityCode, bool isPrivate)
 
 
     }
+
+    public class ServerUtils
+    {
+        private readonly IServer _server;
+
+        public ServerUtils(IServer server)
+        {
+            _server = server;
+        }
+
+        public string GetAddress() {
+            var baseUrl = _server.Features.Get<IServerAddressesFeature>()?.Addresses.FirstOrDefault(a => a.Contains("https"));
+            return baseUrl;
+        }
+    }
```

Comment:
I think this might be best suited in its own file under `...\Utilities\ServerUtils.cs`. It is likely to get used in more places than just this one, that is why we wanted to pull the logic out and put it in its own class.